### DISCUSSION TITLE
Change AM info description to point to am.osowoso.org

### DISCUSSION THREAD

**Discussion by zen0bit** [ivan-hc/AM#563] [Ideas]
![info](https://github.com/ivan-hc/AM/assets/6384793/1e6e22e0-cc97-4e44-b532-5f137d3b3d42)
Here somewhere shoould be mention of am website

**Comment by ivan-hc** [comment]
Before I do this, I need to improve the README... or at least, I need to write an HTML for that. If you see, the URLs are not working and also the menus to expand.

There is a lot of work to do on the README/INDEX

**Comment by ivan-hc** [comment]
@zen0bit https://markdowntohtml.com/

https://am.osowoso.org

there is still some work to do, but in usability its ways better, don't you?

**Comment by ivan-hc** [comment]
Maybe if I use Google Sites as an editor, I can reach the same result as https://bananahackers.net

or https://sites.google.com/view/kaiositalia (the italian version of bananahackers.net)

**Comment by ivan-hc** [comment]
![AM-LOGO](https://github.com/ivan-hc/AM/assets/88724353/0a43b11e-bc94-4f5b-83aa-395bc560d132)

**Comment by ivan-hc** [comment]
![Istantanea_2024-05-03_15-36-44](https://github.com/ivan-hc/AM/assets/88724353/12676c01-0366-461e-9901-d1526bdd2144)

**Comment by ivan-hc** [comment]
I only need to have a structure

**Comment by zen0bit** [comment]
> ![AM-LOGO](https://private-user-images.githubusercontent.com/88724353/327771113-0a43b11e-bc94-4f5b-83aa-395bc560d132.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3NDM3MDUsIm5iZiI6MTcxNDc0MzQwNSwicGF0aCI6Ii84ODcyNDM1My8zMjc3NzExMTMtMGE0M2IxMWUtYmM5NC00ZjViLTgzYWEtMzk1YmM1NjBkMTMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAzVDEzMzY0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExYWE1OGYzN2NiNTNkYjVhOTIzOWViYzM3MTZmNDQ2ZTMzZGZiNDcxZTQxYjBmNmViZjJlZThjOTg1NTMzYWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.G_gDASPOnfkRafe-3A8wQlOo2pdow8Pb4IpsFs_6zX8)

AM looks promising
But application manager is useless there. No one will read it
Or for where is this logo?

I am thinking about favicon now...

**Comment by ivan-hc** [comment]
it's a unique signature, there's no need for others to read it, I've seen lots of AM logos on the web... I wrote this one using gimp/gnome-paint, to distinguish it

**Comment by zen0bit** [comment]
> I only need to have a structure

of?

I leave this up to you, don't know how you want AM site should looks like...
Your choice...

**Comment by ivan-hc** [comment]
I was thinking of giving it the same structure as bananahackers.net, but with fewer pages... more "Kaios italia"

**Comment by zen0bit** [comment]
I like how github kill openning images of others because they cant fix the exploit which was there few years

**Comment by ivan-hc** [comment]
> > I only need to have a structure
> 
> of?
> 
> I leave this up to you, don't know how you want AM site should looks like... Your choice...

I'd like to add 6 pages or urls, to complete the space under the header

And 6 icons

And maybe some music, strictly rock (Nirvana, RATM, AC/DC... I can use one of my [playlists](https://www.youtube.com/@IvanAlexHC86/playlists) )

**Comment by zen0bit** [comment]
> > > I only need to have a structure
> > 
> > 
> > of?
> > I leave this up to you, don't know how you want AM site should looks like... Your choice...
> 
> I'd like to add 6 pages or urls, to complete the space under the header
> 
> And 6 icons
> 
> And maybe some music, strictly rock (Nirvana, RATM, AC/DC... I can use one of my [playlists](https://www.youtube.com/@IvanAlexHC86/playlists) )

Don't see problem with that

**Comment by ivan-hc** [comment]
In this moment I've no idea of what pages I should include.

**Comment by zen0bit** [comment]
> In this moment I've no idea of what pages I should include.

🤪 I thought that you have six, not that you have theme that needs six 🤣

**Comment by ivan-hc** [comment]
maybe I can try another approach

![Istantanea_2024-05-03_16-11-31](https://github.com/ivan-hc/AM/assets/88724353/7a079233-4be5-478b-ab1f-25a14b36e63c)

**Comment by zen0bit** [comment]
PS: I suggest not using spoilers in site
Implementation lack in real world so it wouldn't work for many people...

I recently study this and is better to leave it, until will be really usable across browsers...

PS: Since I personally hate white themes, i suggest use theme with switchable nigh/day black/white

**Comment by ivan-hc** [comment]
sure, I hate it too... now its just a draft. But I always prefer black themes.

**Comment by zen0bit** [comment]
> sure, I hate it too... now its just a draft. But I always prefer black themes.

Why you start with white then...?

**Comment by ivan-hc** [comment]
because the page of Google Sites is white by default

**Comment by ivan-hc** [comment]
![Istantanea_2024-05-03_16-19-30](https://github.com/ivan-hc/AM/assets/88724353/3a3d3d02-6c30-4dfb-b2be-9dd3775d5042)

**Comment by ivan-hc** [comment]
colors and banners will came later

**Comment by ivan-hc** [comment]
but after this, when I'll upload the pages, I need also to remove google's references and trackers from it, obviously

**Comment by zen0bit** [comment]
I don't like that style

Maybe 10 years back OK

**Comment by zen0bit** [comment]
first you dont want any picture behind text.
It will make this less readable

You have to count with both color themes...
So with white and black text must be seen anywhere...

That's no go
It wouldn't work like you expect...

**Comment by ivan-hc** [comment]
I'll not use a Matrix style, bananahackers.net was inspired to the remake of Nokia 8110 used by "Neo" in the "Matrix" movie. Ence the choice of the theme.

> first you dont want any picture behind text. It will make this less readable
> 
> You have to count with both color themes... So with white and black text must be seen anywhere...
> 
> That's no go It wouldn't work like you expect...

I have to think about the colors later, now I don't even have a structure to think about

**Comment by zen0bit** [comment]
Again: Just to not really miss this:

 I suggest not using spoilers in site
Implementation lack in real world so it wouldn't work for many people...

I recently study this and is better to leave it, until will be really usable across browsers...

**Comment by ivan-hc** [comment]
> Again: Just to not really miss this:
> 
> I suggest not using spoilers in site Implementation lack in real world so it wouldn't work for many people...
> 
> I recently study this and is better to leave it, until will be really usable across browsers...

OK

the main problem arises from the fact that [portable-linux-apps.github.io](https://portable-linux-apps.github.io/) is already the very heart of the AM project. AM would be meaningless if there were no installation scripts to install applications. And here another problem arises: the database to be rebuilt from scratch.

It should be included in the new site, or at least embedded. People want AM for apps, not for reading help. A main page with the same content as the README would be a useless duplicate.

**Comment by ivan-hc** [comment]
portable-linux-apps.github.io also includes a JSON file with all applications listed, so for a a good HTML developer would be easy to improve this

**Comment by zen0bit** [comment]
> > Again: Just to not really miss this:
> > I suggest not using spoilers in site Implementation lack in real world so it wouldn't work for many people...
> > I recently study this and is better to leave it, until will be really usable across browsers...
> 
> OK
> 
> the main problem arises from the fact that [portable-linux-apps.github.io](https://portable-linux-apps.github.io/) is already the very heart of the AM project. AM would be meaningless if there were no installation scripts to install applications. And here another problem arises: the database to be rebuilt from scratch.
> 
> It should be included in the new site, or at least embedded. People want AM for apps, not for reading help. A main page with the same content as the README would be a useless duplicate.

Why should be rebuild?

**Comment by ivan-hc** [comment]
For now I'll left it as the readme of the repository. If I get inspired, I'll come back to work on it.

If you have other ideas, feel free to add/edit whatever you want.

**Comment by zen0bit** [comment]
Address is hardcoded everywhere?
Can't database be just copied?

**Comment by ivan-hc** [comment]
> Why should be rebuild?

I had another idea about refactoring of the installation scripts database... but for now its far from reality: to resume all commands on a JSON file or an online database, including all descriptions and commands to use to unstall the apps and create their AM-updaters.

The issue is not much the huge quantity of the installation scripts but to have a structure for that.

**Comment by ivan-hc** [comment]
> Address is hardcoded everywhere? Can't database be just copied?

in what sense?

**Comment by zen0bit** [comment]
> > Why should be rebuild?
> 
> I had another idea about refactoring of the installation scripts database... but for now its far from reality: to resume all commands on a JSON file or an online database, including all descriptions and commands to use to unstall the apps and create their AM-updaters.
> 
> The issue is not much the huge quantity of the installation scripts but to have a structure for that.

Which structure you mean?
or better to say structure of what?

**Comment by zen0bit** [comment]
You don't want everything in one JSON file...

**Comment by ivan-hc** [comment]
> Which structure you mean?

One file? More files?

How should each file be compiled to resume and mege together a message (for -a), the download urls and the extraction (-i) and update (-u)?

I'd like to have a database with the updated urls for each application, so the installation scripts should no more rely on what is hosted but tho the still working url, see for example https://github.com/ivan-hc/AM/issues/539

**Comment by ivan-hc** [comment]
again, this is far from reality for now... its just a fantasy of mine

but this is an off topic

**Comment by zen0bit** [comment]
One template per app
as I said
 
As simple as posible.
Main power will be that anyone can add own AppImage.

So It must be really simple..

**Comment by zen0bit** [comment]
Main question is how much from installation scripts differ between apps.

Just theoreticaly:
Sort them to (from github, sourceforge, gitlab, gitea, ... , own)

If you still have most apps in own categories we will have problems. If you have just few we are good.

**Comment by ivan-hc** [comment]
more than an app template... a reference URL for download. See here, from the lmstudio script I've fixed yesterday:
```
version=$(mkdir -p "$HOME"/.cache && cd "$HOME"/.cache && wget -rq https://lmstudio.ai 2>&1 && cat ./lmstudio.ai/static/js/main*.js | tr '"' '\n' | grep -i appimage && rm -Rf "$HOME"/.cache/lmstudio.ai)
```
I had to change it because the download page has been changed. This approach will slow down also the update, because the script must download and check an hidden url into a deep js file.

The approach I'd like to have is a file that lists all download urls directly. Github actions can take case of updating these.

No more missing urls this way. We have a snapshot for the apps and people can still have the apps they want, also if not updated... but better than nothing.

**Comment by ivan-hc** [comment]
> Main question is how much from installation scripts differ between apps.
> 
> Just theoreticaly: Sort them to (from github, sourceforge, gitlab, gitea, ... , own)
> 
> If you still have most apps in own categories we will have problems. If you have just few we are good.

also this is a big issue, use the command
```
am -d platform-tools brave firefox-nightly nootka node python3.9 coreuniverse inkscape blender kdegames libreoffice wine kdenlive ffwa-whatsapp windows95
```
they are just samples

**Comment by zen0bit** [comment]
> No more missing urls this way. We have a snapshot for the apps and people can still have the apps they want, also if not updated... but better than nothing. Not understand this..

Don't undestand this.

CI 👍

**Comment by ivan-hc** [comment]
> Don't undestand this.

The ChaoticAUR, for example, already compiles the packages from AUR and then stores the package in its repos. If a PKGBUILD is broken, ChaoticAUR still have a working URL for the ready binary.

The same way, I'd like to have a snapshot of the existing URLs, all the installation scripts should only rely on the URL listed. So what happened at https://github.com/ivan-hc/AM/issues/539 should no more happen, at least to have a package, and not nothing.

**Comment by zen0bit** [comment]
> ```
> am -d platform-tools brave firefox-nightly nootka node python3.9 coreuniverse inkscape blender kdegames libreoffice wine kdenlive ffwa-whatsapp windows95
> ```
> 
> they are just samples

Problem?

**Comment by ivan-hc** [comment]
run the command, download these scripts and compare them

EDIT: are you afraid to use the -d option because on your fork something bay be broken? Better! You would understand more how I feel when I take tests. xD

**Comment by ivan-hc** [comment]
> run the command, download these scripts and compare them
> 
> EDIT: are you afraid to use the -d option because on your fork something bay be broken? Better! You would understand more how I feel when I take tests. xD

OK, I was right.

@zen0bit R.I.P.
